src/decouplededitor.js

```diff
@@ -107,18 +133,107 @@ export default class DecoupledEditor extends Editor {
+	 * have to be a string either. You can e.g. return an object or a DOM `DocumentFragment`  if you consider this
+	 * the right format for you.
+	 *
+	 * @method #getData
+	 * @param {String} [rootName='main'] The root name from which to get data.
+	 * @returns {String} Output data.
+	 */
+	getData( rootName = 'main' ) {
+		if ( !this._rootNames.includes( rootName ) ) {
+			/**
+			 * Thrown when there is an attempt to get data on a non-existing root.
+			 *
+			 * @error trying-to-get-data-on-non-existing-root
+			 */
+			throw new CKEditorError( 'trying-to-get-data-on-non-existing-root: ' +
+				`Attempting to get data from non-existing "${ rootName }" root.`
+			);
+		}
+
+		return this.data.get( rootName );
+	}
+
+	/**
+	 * Sets the data in the editor editable areas.
+	 *
+	 * There are couple of ways `setData` can be used. When only data is passed it is set on the default root (`main`):
+	 *
+	 *		editor.setData( '<h1>CKEditor 5</h1>' ); // Sets data on the `main` root.
+	 *
+	 * Along with the data, object specifying root name can be passed:
+	 *
+	 *		editor.setData( '<p>This is editor!</p>', { rootName: 'content' } ); // Sets data on the `content` root.
+	 *
+	 * Finally, object specifying `rootName` - `data` pairs could be used:
+	 *
+	 *		editor.setData( { main: '<h1>CKEditor 5</h1>', content: '<p>This is editor!</p>' } ); // Sets data on `main` and `content` root.
+	 *
+	 * By default the editor accepts HTML. This can be controlled by injecting a different data processor.
+	 * See the {@glink features/markdown Markdown output} guide for more details.
+	 *
+	 * Note: Not only is the format of the data configurable, but the type of the `setData()`'s data values does not
+	 * have to be a string either. The data can be e.g. an object or a DOM `DocumentFragment` if you consider this
+	 * the right format for you.
+	 *
+	 * @method #setData
+	 * @param {String|Object.<String,String>} data Data as a sting or an object containing input data where each key
+	 * is the name of the root and corresponding value is a data to set.
+	 * @param {Object} [options] The options object specifying on which root to set data. Has effect only if first
+	 * parameter is passed as a string.
+	 */
+	setData( data, options ) {
+		let roots = {};
+
+		if ( typeof data === 'string' ) {
+			const rootName = options && options.rootName ? options.rootName : 'main';
+			roots[ rootName ] = data;
+		} else {
+			roots = data;
+		}
+
+		for ( const [ rootName, content ] of Object.entries( roots ) ) {
+			if ( !this._rootNames.includes( rootName ) ) {
+				/**
+				 * Thrown when there is an attempt to set data on a non-existing root.
+				 *
+				 * @error trying-to-set-data-on-non-existing-root
+				 */
+				log.warn( `trying-to-set-data-on-non-existing-root: Attempting to set data on non-existing "${ rootName }" root.` );
+			} else {
+				this.data.set( content, rootName );
```

Comment:
Since you already changed `data.init` you should also change `data.set` in the same way, so this logic should be moved to `data.set` method.